include/mbedtls/ecp.h

```diff
@@ -503,6 +503,58 @@ int mbedtls_ecp_tls_read_group( mbedtls_ecp_group *grp, const unsigned char **bu
 int mbedtls_ecp_tls_write_group( const mbedtls_ecp_group *grp, size_t *olen,
                          unsigned char *buf, size_t blen );
 
+#if defined(MBEDTLS_ASN1_WRITE_C) && defined(MBEDTLS_OID_C)
+/**
+ * \brief           Maximum size of the output of mbedtls_ecp_ansi_write_group
+ *
+ * \note            The maximum size of the OID of a supported group + 2 for
+ *                  tag and length. Maximum size 30 is based on the length of
+ *                  the OID for primeCurves 10-38 over GF(p) defined by the
+ *                  CDC Group, as they seem to have the longest OID out of
+ *                  curves in use.
+ */
+#define MBEDTLS_ECP_GRP_OID_MAX_SIZE ( 30 + 2 )
+
+/**
+ * \brief           Write the ANSI X9.62/RFC5480 OID ECParameters of a group
+ *
+ *                  The output is the group's OID wrapped as ASN.1.
+ *
+ * \param grp       ECP group used
+ * \param p         Buffer to write to
+ * \param size      Buffer size
+ * \param olen      Number of bytes written to \c buf
+ * \return          0 on success
+ *                  or \c MBEDTLS_ERR_ECP_BUFFER_TOO_SMALL
+ *                  or \c MBEDTLS_ERR_OID_NOT_FOUND
+ */
+int mbedtls_ecp_ansi_write_group( const mbedtls_ecp_group *grp,
```

Comment:
The TLS encoding (used by the preexisting functions `mbedtls_ecp_tls_{read,write}_{group,point}`) is also an ASN.1 encoding. I think ASN.1 is too vague. If not “ANSI”, we could describe it as “X9.62” or “RFC5480” or maybe even “X.509”. But what's wrong with `ansi`?

Reply:
Nothing, I just thought it would be more intuitive like that. But I see your point, I wasn't taking the preexisting functions and you are right, calling it `_asn1_` wouldn't do any good.

It didn't occur to me, but now that you mention it I feel that `_x509_` would tell more about the use of this function than `_ansi_`.

Reply:
@gilles-peskine-arm TLS encoding is not ASN.1, TLS uses its own custom data description language and associated encoding (for the 4 functions you mention). However, I agree that ASN.1 is too vague, as it's the name of a description language (and in the context of mbed TLS, implicitly we also mean the associated DER or BER encodings), and what we mean here is a particular structure/encoding defined by the standard using that language.

@yanesca I think ANSI is OK though a bit vague (the whole ECP module could be argued to be based on ANSI X9.62, even though in practice I used SEC1 instead because it's publicly accessible), and while `x509` might seem more meaningful, technically it's too restrictive as this encoding is also use outside of X.509 (for example, when saving keys in files, or here, communicating with HSMs).

At first I was surprised by the `ansi` in the name (my brain even interpreted it as a typo for `asn1` for a few seconds), but I wasn't able to come up with a better name. I think we'll have to settle for an OK name and use the documentation to clarify that the difference between the various "write foo" functions are.

Reply:
Yes, I had second thoughts about `_x509_` after I wrote that. I can't think of anything better either and happy to settle for `_ansi_`.